### PR TITLE
fixes issue-1180 fixes the post padding for mobile devices found in timeline.jsx

### LINKED ISSUE #1180: Remove outside post padding in mobile
**What would you like to be added**:
After #1174 was pushed, posts on mobile have padding, which prevents them from taking up the full screen. 

What we have:
image

What we want:
image

**Why would you like this to be added**:
The padding makes the paragraphs more narrow than they need to be, allowing for less words per line.

### LINKED ISSUE #1226: Fix timeline width in search results
In #1217 I updated how our search results show for posts, but I couldn't get the width right on the container (there's extra padding/margin that makes it narrower than it should be):

Screen Shot 2020-10-20 at 9 52 59 PM

I think the posts should be the same width as the search UI above it.

### PR DESCRIPTION
**Pull request by pyvelkov**
## Issue This PR Addresses
Fixes #1180 

## Type of Change

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Description
I traced how the padding was being applied through Chrome developer tools. I saw that the default root `padding` was being applied meaning the posts have `padding` to the left and right and were not being overridden as expected. From the previous fix that was made I saw that root style was being overridden with a `padding` of 0 in posts.jsx but since the change in #1217 , timeline.jsx was created and that is where the issue stemmed from. I saw that the Timeline function was returning the wrapped array in a container that had the `className={classes.root}`. However custom overriding of the root style was not being called in "useStyles". 

### Before
without fix

### After
with fix

<br /><br />
EDIT:
I believe this also fixed #1226 
fixed search

I also ran `npm run test` and everything passed. Everything looks alright on my end.

Let me know what you think and if this is what we are after in #1180 , my JS is terrible and I probably spent far longer than needed for something like this. I hope it is what we are after.

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by humphd** [comment]
I'm not sure if we should deal with this here or in another issue, but I also notice we have an overflow problem on mobile with content extending beyond the parent container(s):

width-bug

**Comment by pyvelkov** [comment]
> I'm not sure if we should deal with this here or in another issue, but I also notice we have an overflow problem on mobile with content extending beyond the parent container(s):
> 
> width-bug

I don't mind doing this in another post but I can do it here. I sort of found a fix by applying style of`overflown` content to `auto` and it added a scroll like effect for posts like this.
scroll on certain posts

**Comment by humphd** [comment]
Sure, push another commit and let's do it here.

**Comment by pyvelkov** [comment]
> Sure, push another commit and let's do it here.

I just added the commit that adds the scrollbar for overflown posts. Let me know, Thanks :)

**Review by cindyledev — DISMISSED** [review]
Thanks a lot for fixing this

File: src/frontend/src/components/Post/Post.jsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { makeStyles } from '@material-ui/core/styles';
 import { Box, Grid, Typography, ListSubheader } from '@material-ui/core';
 import './telescope-post-content.css';
 import AdminButtons from '../AdminButtons';
+import { autoDetection } from 'highlight.js';
```
Comment (c3ho) [inline src/frontend/src/components/Post/Post.jsx:9]: Is it just me? I don't see this being used anywhere in the code change.

**Review by c3ho — CHANGES_REQUESTED** [review]

File: src/frontend/src/components/Post/Post.jsx (hunk 1 above)
Comment (manekenpix) [reply src/frontend/src/components/Post/Post.jsx:9]: we have this:
```js
import 'highlight.js/styles/github.css';
```
in line 4. So what is this in line 9 for?

Comment (pyvelkov) [reply src/frontend/src/components/Post/Post.jsx:9]: oh shoot. I think I was testing a fix for the overflown content mentioned above. I will remove it. I apologize.

Comment (pyvelkov) [reply src/frontend/src/components/Post/Post.jsx]: Good catch @c3ho I will check for that in the future. Sorry about that again.

**Review by c3ho — APPROVED** [review]

**Review by manekenpix — APPROVED** [review]

**Comment by c3ho** [comment]
@pyvelkov Feel free to squash and merge

**Comment by pyvelkov** [comment]
> @pyvelkov Feel free to squash and merge

would you like me to do it, or..?

**Comment by humphd** [comment]
@pyvelkov go for it!  With 2 reviews giving an approval, you can squash and merge.

**Pull request merged**